src/coreclr/nativeaot/docs/containers.md

```diff
@@ -0,0 +1,121 @@
+# Building native AOT apps in containers
+
+Containers are a useful tool to build any kind of code, including native AOT. There are a few patterns for doing that.
+
+## Cloud native apps
+
+Native AOT apps can be built in containers (with multi-stage build) just like regular (CoreCLR) apps. SDK `aot` images provide the additional native toolchain dependencies required to build native AOT apps.
+
+Search for `aot` images at [dotnet/nightly/sdk](https://mcr.microsoft.com/en-us/product/dotnet/nightly/sdk/tags).
+
+The [`releasesapi`](https://github.com/dotnet/dotnet-docker/blob/main/samples/releasesapi/README.md) sample demonstrates how to build native AOT apps in containers.
+
+## Cross-compiling
+
+For cloud native apps, build and runtime OS typically match, at least if you use multi-stage build. Once you step out of containers (for app delivery), it is more likely that you are delivering binaries that you want to work in more places (like on older Linux distros).
```

Comment:
> Once you step out of containers (for app delivery), it is more likely that you are delivering binaries that you want to work in more places (like on older Linux distros).

I agree. I would keep the sample tailored to this exact scenario by using console app. Console apps are the most likely target for binary delivery. We should not be encouraging people to publish cloud services this way.

Reply:
The key reason I did that is because I had a ready sample. I'm happy to use a different one. Do you have a good console app sample I can use?

Reply:
It can be console app created on the fly for demo purposes:

First build image.
```
docker build --pull -t cross-build -f Dockerfile.cross-build-x64-x64 . 
```
Create an empty console app for demo purposes.
```
mkdir myapp
dotnet new console --aot
```
Build the app, with volume mounting.
```bash
docker run --rm -it -v .:/source -w /source cross-build dotnet publish -o . -p:SysRoot=/crossrootfs/x64 -p:LinkerFlavor=lld myapp.csproj
```
Finally, verify that the app works on Ubuntu 16.04 (the oldest currently supported Ubuntu version):

```bash
docker run --rm -it -v .:/myapp ubuntu:16.04 myapp/myapp
```

Reply:
That's a much better idea.